packages/block-library/src/list/split-on-line-break.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * External dependencies
+ */
+import { last } from 'lodash';
+
+/**
+ * Split the content of a paragraph on line breaks ('<br>') into sets of
+ * content, each representing a list item.
+ *
+ * The term "content" refers to a rich-text description in the form of a mixed
+ * array of strings and element-like objects (e.g. `{ type: strong, props: {
+ * children: […] } }`).
+ *
+ * @param {Array}    fragments Rich-text content
+ * @return {Array}   Array of rich-text content
+ */
+export default function splitOnLineBreak( fragments ) {
```

Comment:
I think we have the objective of encapsulating details about the children object (it has a type, it has props etc...) inside packages/blocks/src/api/children.js. Would it make sense to implement this function there?

Reply:
Hm, I had left this note in the PR description:

> - I wrote `splitOnLineBreak` as a single file inside the List block's source directory to make it easier to test.
> - I didn't place it in a utils module because I don't necessarily want this to be a public module to manipulate rich text.

Does it make sense to you? Should we later revise what we'd like to exist in `api/children`?

Reply:
Yes it makes sense, we have the downside of accessing the children object directly but I don't see a better alternative.
The alternative I see would be using toHTML to convert the children to HTML manipulate the HTML and then convert it back to a children object using fromDOM, but that seems like an unnecessary complexity and would make the code much more complex.
I think give the situation accessing the props of the children object here seems to be the best option. Maybe in the future, we can think of moving the function to api/children or adding abstractions in api/children that make this function possible.